stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/channels/MessageReadStatusIcon.kt

```diff
@@ -11,51 +11,76 @@ import io.getstream.chat.android.client.models.Message
 import io.getstream.chat.android.compose.previewdata.PreviewMessageData
-import io.getstream.chat.android.compose.previewdata.PreviewUserData
 import io.getstream.chat.android.compose.ui.theme.ChatTheme
 import io.getstream.chat.android.compose.ui.util.getReadStatuses
 
 /**
- * Shows the last message read status for each channel item based on the given
- * information about the channel, the last message and the current user.
+ * Shows a delivery status indicator for a particular message.
  *
- * @param channel The channel to show the read status for.
- * @param lastMessage The last message in the channel.
- * @param currentUser Currently logged in user.
+ * @param channel The channel with channel reads to check.
+ * @param message The message with sync status to check.
+ * @param currentUser The currently logged in user.
  * @param modifier Modifier for styling.
  */
 @Composable
 public fun MessageReadStatusIcon(
     channel: Channel,
-    lastMessage: Message,
+    message: Message,
     currentUser: User?,
     modifier: Modifier = Modifier,
 ) {
     val readStatues = channel.getReadStatuses(userToIgnore = currentUser)
-    val syncStatus = lastMessage.syncStatus
-    val currentUserSentMessage = lastMessage.user.id == currentUser?.id
-    val readCount = readStatues.count { it.time >= lastMessage.getCreatedAtOrThrow().time }
+    val readCount = readStatues.count { it.time >= message.getCreatedAtOrThrow().time }
+    val isMessageRead = readCount != 0
 
-    val messageIcon = when {
-        currentUserSentMessage && readCount == 0 -> R.drawable.stream_compose_message_sent
-        !currentUserSentMessage || readCount != 0 -> R.drawable.stream_compose_message_seen
-        syncStatus == SyncStatus.SYNC_NEEDED || syncStatus == SyncStatus.AWAITING_ATTACHMENTS -> R.drawable.stream_compose_ic_clock
-        syncStatus == SyncStatus.COMPLETED -> R.drawable.stream_compose_message_sent
-        else -> null
-    }
+    MessageReadStatusIcon(
+        message = message,
+        isMessageRead = isMessageRead,
+        modifier = modifier,
+    )
+}
 
-    if (messageIcon != null) {
-        val iconTint =
-            if (!currentUserSentMessage || (readStatues.isNotEmpty() && readCount == readStatues.size)) ChatTheme.colors.primaryAccent else ChatTheme.colors.textLowEmphasis
+/**
+ * Shows a delivery status indicator for a particular message.
+ *
+ * @param message The message with sync status to check.
+ * @param isMessageRead If the message is read by any member.
+ * @param modifier Modifier for styling.
+ */
+@Composable
+public fun MessageReadStatusIcon(
+    message: Message,
+    isMessageRead: Boolean,
+    modifier: Modifier = Modifier,
+) {
+    val syncStatus = message.syncStatus
 
-        Icon(
-            modifier = modifier,
-            painter = painterResource(id = messageIcon),
-            contentDescription = null,
-            tint = iconTint,
-        )
+    when {
```

Comment:
Idk if this is less code but maybe sth like:

```
val color = when {
  isMessageRead ->
  syncStatusCheck ->
  syncStatusCheck2 ->
}

val icon = when { 
...
}

Icon(
  modifier = modifier,
  icon = icon,
  tint = tint,
  contentDescription = null
)
```

Reply:
There are also other statuses like "failed permanently" that we display without delivery indicator at all. So, the snippet above will have an "if" check before the Icon.

Reply:
yeah that sounds good - if you feel like it's a bit more readable or sth we can go in that direction, if it's not, then I'm fine with the current impl.

Reply:
👍